### PR TITLE
native client test for hibernate

### PR DESCRIPTION
**Pull request by mustafaiman**

### CODE REVIEW

File: hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+@Category({QuickTest.class, ParallelTest.class})
+public class NativeClientTest
+        extends HibernateSlowTestSupport {
+
+    protected SessionFactory clientSf;
+
+    @Override
+    protected Properties getCacheProperties() {
+        Properties props = new Properties();
+        props.setProperty(Environment.CACHE_REGION_FACTORY, HazelcastCacheRegionFactory.class.getName());
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java]: do we need to clear this inside `@After`?

File: hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        session = clientSf.openSession();
+        DummyEntity retrieved = (DummyEntity) session.get(DummyEntity.class, (long)1);
+        assertEquals("dummy:1", retrieved.getName());
+    }
+
+    @After
+    public void tearDown() {
+        if(clientSf !=null) {
+            clientSf.close();
+            clientSf = null;
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java]: I don't see any reason to set it to `null`.

**Comment by mesutcelik** [comment]
two minor comments :+1:

File: hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java (hunk 1 above)
Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/NativeClientTest.java]: this is not an environment variable. it is passed to hibernate on creation. it is gone once hibernate's session factory is closed

**Comment by kobalski** [comment]
:+1:  Merging, but will change these tests as slow test with direct commit, because hibernate tests which does not use mock network must be slow test.

**Pull request merged**